src/stratis_cli/_actions/_top.py

```diff
@@ -71,25 +105,39 @@ def _check_same_tier(managed_objects, to_be_added, this_tier):
     :type managed_objects: dict of str * dict
     :param to_be_added: the blockdevs to be added
     :type to_be_added: frozenset of str
-    :param other_tier: the tier requested
-    :type other_tier: _stratisd_constants.BlockDevTiers
+    :param this_tier: the tier requested
+    :type this_tier: _stratisd_constants.BlockDevTiers
     :raises StratisCliPartialChangeError: if blockdevs are used by this tier
+    :raises StratisCliInUseSameTierError: if blockdevs are used by this tier in another pool
     """
     # pylint: disable=import-outside-toplevel
     from ._data import MODev
     from ._data import devs
 
-    these = frozenset(
-        str(MODev(info).Devnode())
-        for (_, info) in devs(props={"Tier": this_tier}).search(managed_objects)
+    pools_to_blockdevs = _generate_pools_to_blockdevs(
+        managed_objects, to_be_added, this_tier
+    )
+
+    owned_by_current_pool = frozenset(
```

Comment:
Probably too complicated...I think you should just be able to do a get in pools_to_blockdevs, like:

```frozenset(pools_to_blockdevs.get(pool_name, []))```

Reply:
Resolved.